src/github_app.py

```diff
@@ -0,0 +1,50 @@
+"""
+This module contains the logic to support running the app as a Github App
+"""
+import time
+
+import jwt
+import requests
+
+
+class GithubAppToken:
+    def __init__(self, private_key, app_id, installation_id) -> None:
+        self.headers = self.get_authentication_header(private_key, app_id)
+        self.installation_id = installation_id
+
+    # From docs: Installation access tokens have the permissions
+    # configured by the GitHub App and expire after one hour.
+    def get_token(self):
+        req = requests.post(
+            url=f"https://api.github.com/app/installations/{self.installation_id}/access_tokens",
+            headers=self.headers,
+        )
+        req.raise_for_status()
+        resp = req.json()
+        # This token expires in an hour
+        return resp["token"]
+
+    # This is executed when we enter a with block
+    def __enter__(self):
+        return self.get_token()
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        requests.delete("https://api.github.com/installation/token")
```

Comment:
here's what I mean -- also I fixed the token deletion since it was missing the auth

```suggestion
    # From docs: Installation access tokens have the permissions
    # configured by the GitHub App and expire after one hour.
    @contextlib.contextmanager
    def get_token(self) -> Generator[str, None, None]:
        req = requests.post(
            url=f"https://api.github.com/app/installations/{self.installation_id}/access_tokens",
            headers=self.headers,
        )
        req.raise_for_status()
        resp = req.json()
        try:
            # This token expires in an hour
            yield resp["token"]
        finally:
            requests.delete("https://api.github.com/installation/token", headers={"Authorization": f"token {resp['token']}"})
```